### PR TITLE
Small fixes 2

### PR DESCRIPTION
**Pull request by alex-mckenna**
## Still TODO:

  - [ ] Write a changelog entry (see changelog/README.md)
  - [x] Check copyright notices are up to date in edited files

[comment]: # (Depending on the change, some of these points may not be necessary. If so, leave the boxes unchecked so it is easier for a reviewer to see what has been omitted.)

### CODE REVIEW

File: clash-lib/src/Clash/Core/TermInfo.hs [hunk 1]
```diff
@@ -183,7 +183,8 @@ piResultTyMaybe m ty arg
   | Just ty' <- coreView1 m ty
   = piResultTyMaybe m ty' arg
   | FunTy a res <- tyView ty
-  = if debugIsOn && not (aeqType a arg) then error [I.i|
+  -- TODO Remove when coreview is not here: types should be aeq anyway
```
Comment (martijnbastiaan) [inline clash-lib/src/Clash/Core/TermInfo.hs:186]: I don't get what this comment is trying to convey. What needs to be removed?

File: clash-lib/src/Clash/Core/Type.hs [hunk 2]
```diff
@@ -251,6 +252,7 @@ coreView1 tcMap ty = case tyView ty of
          AlgTyCon {algTcRhs = (NewTyCon _ nt)}
            -> newTyConInstRhs nt args
          _ -> reduceTypeFamily tcMap ty
+  OtherType (AnnType _ ty') -> coreView1 tcMap ty'
```
Comment (martijnbastiaan) [inline clash-lib/src/Clash/Core/Type.hs:255]: This probably needs a changelog (albeit under "internal changes")

File: clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs [hunk 3]
```diff
@@ -733,7 +733,7 @@ ghcPrimStep tcm isSubj pInfo tys args mach = case primName pInfo of
            mbaTy = mkFunTy intPrimTy (last tyArgs)
            newE = mkApps (Data tupDc) (map Right tyArgs ++
                     [Left (Prim rwTy)
-                    ,Left (mkApps (Prim (PrimInfo "GHC.Prim.MutableByteArray#" mbaTy WorkNever SingleResult))
+                    ,Left (mkApps (Prim (PrimInfo "GHC.Prim.MutableByteArray#" mbaTy WorkNever SingleResult Nothing))
```
Comment (martijnbastiaan) [inline clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs:736]: `Nothing` is quite indiscriptive. Could you change this to initializing with field names?

File: clash-lib/src/Clash/Normalize/Transformations.hs [hunk 4]
```diff
@@ -25,3 +25,4 @@ import Clash.Normalize.Transformations.Reduce as X
 import Clash.Normalize.Transformations.SeparateArgs as X
 import Clash.Normalize.Transformations.Specialize as X
 import Clash.Normalize.Transformations.XOptimize as X
+
```
Comment (martijnbastiaan) [inline clash-lib/src/Clash/Normalize/Transformations.hs:28]: nice diff :P

**Review by martijnbastiaan — CHANGES_REQUESTED** [review]

File: clash-lib/src/Clash/Core/TermInfo.hs (hunk 1 above)
Comment (alex-mckenna) [reply clash-lib/src/Clash/Core/TermInfo.hs:186]: The thing, when the other thing isn't, _what's so hard to understand?_ /s

**Pull request merged**